src/uct/ib/mlx5/dv/ib_mlx5_dv.c

```diff
@@ -72,15 +103,9 @@ ucs_status_t uct_ib_mlx5_devx_create_qp(uct_ib_iface_t *iface,
         goto err;
     }
 
-    wqe_size = sizeof(struct mlx5_wqe_ctrl_seg) +
-               sizeof(struct mlx5_wqe_umr_ctrl_seg) +
-               sizeof(struct mlx5_wqe_mkey_context_seg) +
-               ucs_max(sizeof(struct mlx5_wqe_umr_klm_seg), 64) +
-               ucs_max(attr->super.cap.max_send_sge * sizeof(struct mlx5_wqe_data_seg),
-                       ucs_align_up(sizeof(struct mlx5_wqe_inl_data_seg) +
-                                    attr->super.cap.max_inline_data, 16));
-    len_tx = ucs_roundup_pow2_or0(attr->super.cap.max_send_wr * wqe_size);
-    max_tx = len_tx / MLX5_SEND_WQE_BB;
+    max_tx = uct_ib_mlx5_sq_length(md, attr->super.cap.max_send_wr,
+                                   attr->super.qp_type);
```

Comment:
can QP type be `IBV_QPT_RC` and no devx? `uct_ib_mlx5_sq_length` should abort then

Reply:
yes, we have a `ucs_fatal()` there to cover such case

Reply:
no, i meant that this does not seem to be right. This function has quite general name, can we add some default for RC QP type as well (or for any other qp type instead of fatal)? We can print a diag/debug message there also

Reply:
yes, we could update the function to return `config->super.tx_cq_len` for RC